src/hooks/useHandleRedeem.tsx

```diff
@@ -339,45 +343,72 @@ async function solana(
   wallet: SolanaWallet,
   payerAddress: string, //TODO: we may not need this since we have wallet
   signedVAA: Uint8Array,
-  isNative: boolean
+  isNative: boolean,
+  isTbtc: boolean
 ) {
   dispatch(setIsRedeeming(true));
   try {
     if (!wallet.signTransaction) {
       throw new Error("wallet.signTransaction is undefined");
     }
     const connection = new Connection(SOLANA_HOST, "confirmed");
-    await postVaaSolanaWithRetry(
-      connection,
-      wallet.signTransaction.bind(wallet),
-      SOL_BRIDGE_ADDRESS,
-      payerAddress,
-      Buffer.from(signedVAA),
-      MAX_VAA_UPLOAD_RETRIES_SOLANA
-    );
-    // TODO: how do we retry in between these steps
-    const transaction = isNative
-      ? await redeemAndUnwrapOnSolana(
-          connection,
-          SOL_BRIDGE_ADDRESS,
-          SOL_TOKEN_BRIDGE_ADDRESS,
-          payerAddress,
-          signedVAA
-        )
-      : await redeemOnSolana(
-          connection,
-          SOL_BRIDGE_ADDRESS,
-          SOL_TOKEN_BRIDGE_ADDRESS,
-          payerAddress,
-          signedVAA
-        );
-    const txid = await signSendAndConfirm(wallet, transaction);
-    // TODO: didn't want to make an info call we didn't need, can we get the block without it by modifying the above call?
-    dispatch(setRedeemTx({ id: txid, block: 1 }));
-    enqueueSnackbar(null, {
-      content: <Alert severity="success">Transaction confirmed</Alert>,
-    });
+    if (isTbtc) {
+      // TODO compute the amount of tx that postVaaSolanaWithRetry
+      // will create to notice the user up front
+      // we could call createPostSignedVaaTransactions to create fake txs
+      // and read the length of the array
+      await postVaaSolanaWithRetry(
+        connection,
+        wallet.signTransaction.bind(wallet),
+        SOL_BRIDGE_ADDRESS,
+        payerAddress,
+        Buffer.from(signedVAA),
+        MAX_VAA_UPLOAD_RETRIES_SOLANA
+      );
```

Comment:
This is done here (in the `if` case) and again in the `else` case, so maybe we could move it outside the `if else` statement? since ti seems we always need to do this.

Reply:
yes, good call